### PR TITLE
Specify ruby version in android build workflow

### PR DESCRIPTION
**Pull request by roryabraham**
This is an attempt to resolve the [Android build failures](https://github.com/Expensify/Expensify.cash/runs/1703069542?check_suite_focus=true) we're seeing on production.

### Details
[Ruby `3.0`](https://www.ruby-lang.org/en/downloads/releases/) was release on Christmas, 2020. It's a new LTS version, but Ruby is not backwards compatible. One of the dependencies listed in our `Gemfile.lock` is `google-api-client-0.38.0`, which requires Ruby 2.4. To fix this, I did two things:

1) Ran `bundle update` to update our ruby dependencies
2) `Gemfile.lock` still specified `google-api-client-0.38.0`, so I added a ruby version specification to [actions/setup-ruby](https://github.com/actions/setup-ruby). Note that this was done only on Android, which is the only platform on which we seem to see this error. Not sure why that would be, considering all platforms perform the actions which have broken on Android.

### Fixed Issues
n/a, just trying to get our production builds working.

### Tests
In order to test this, it must be merged.

### CODE REVIEW

**Comment by roryabraham** [comment]
Using ruby 2.7 should be fine, because the error says:

> google-api-client-0.38.0 requires ruby version ~> 2.4, which is incompatible
with the current version, ruby 3.0.0p0

But I looked into that tidle notation called [pessimistic version constraint](https://stackoverflow.com/questions/3414337/meaning-of-tilde-greater-than-in-version-requirement) and 2.7 should be fine.

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
             - uses: actions/checkout@v2
 
             - uses: actions/setup-ruby@v1
+              with:
+                  ruby-version: '2.7'
```
Comment (AndrewGable) [inline .github/workflows/android.yml:17]: Should we lock down all the other usages of `setup-ruby`?

**Review by AndrewGable — COMMENTED** [review]
We use `rvm` in other git repos, it might be interesting to see if setting a `.ruby-version` in this repo would solve this like we have elsewhere? e.g. https://rvm.io/workflow/projects#project-file-ruby-version

File: .github/workflows/android.yml (hunk 1 above)
Comment (roryabraham) [reply .github/workflows/android.yml:17]: I would assume that the same error could occur on any of the workflows that are using setup-ruby and bundler. Probably a good idea to do the same in all the other workflows.

**Comment by roryabraham** [comment]
I'm not sure about specifying `.ruby-version`. I think it would probably work, but for now I think we should try this method because it's clearly described in the documentation for `actions/setup-ruby`

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
Merging to unblock deploys

**Pull request merged**